certora/specs/RevertsMintBurnEthereum.spec

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+methods {
+    function owner() external returns address envfree;
+    function totalSupply() external returns uint256 envfree;
+    function balanceOf(address) external returns uint256 envfree;
+
+    // Avoids checking delegation related overflows.
+    function _._moveDelegateVotes(address, address, uint256) internal => CONSTANT;
+}
+
+// Check that minting to zero address or minter is not owner revert.
+rule mintReverts(env e, address to, uint256 amount) {
+    // Safe require as mint is a non-payable function.
+    require e.msg.value == 0;
+
+    require totalSupply() + amount <= max_uint256;
```

Comment:
This is also a valid revert condition: you cannot mint more than max_uint256

Reply:
Not exactly, in OZ's code operation is done in an `unchecked` block.